firebase-firestore/src/main/java/com/google/firebase/firestore/util/BackgroundQueue.java

```diff
@@ -40,8 +42,12 @@ public void execute(Runnable task) {
   }
 
   /** Wait for all currently scheduled tasks to complete. */
-  public void drain() throws InterruptedException {
-    completedTasks.acquire(pendingTaskCount);
-    pendingTaskCount = 0;
+  public void drain() {
+    try {
+      completedTasks.acquire(pendingTaskCount);
+      pendingTaskCount = 0;
+    } catch (InterruptedException e) {
+      fail("Interrupted while waiting for background task", e);
```

Comment:
It's not a good idea to swallow `InterruptedException`. You should at least call `Thread.currentThread().interrupt()` before calling `fail()`.

https://www.baeldung.com/java-interrupted-exception

Reply:
Added Thread.currentThread().interrupt() (also recommended by other literature)